tests/integration/test_config.py

```diff
@@ -0,0 +1,101 @@
+        {
+            "logging_log_min_duration_statement": {"2147483648", "1"}
+        },  # config option is between -1 and 2147483647
+        {
+            "memory_maintenance_work_mem": {"2147483648", "1024"}
+        },  # config option is between 1024 and 2147483647
+        {"memory_max_prepared_transactions": {"-1", "0"}},  # config option is between 0 and 262143
+        {"memory_shared_buffers": {"15", "16"}},  # config option is greater or equal than 16
+        {"memory_temp_buffers": {"99", "100"}},  # config option is between 100 and 1073741823
+        {"memory_work_mem": {"2147483648", "64"}},  # config option is between 64 and 2147483647
+        {
+            "optimizer_constraint_exclusion": {test_string, "off"}
+        },  # config option is one of `on`, `off` or `partition`
+        {
+            "optimizer_default_statistics_target": {"0", "1"}
+        },  # config option is between 1 and 10000
+        {"optimizer_from_collapse_limit": {"0", "1"}},  # config option is between 1 and 2147483647
+        {"optimizer_join_collapse_limit": {"0", "1"}},  # config option is between 1 and 2147483647
+        {"profile": {test_string, "testing"}},  # config option is one of `testing` or `production`
+        {"profile_limit_memory": {"127", "128"}},  # config option is between 128 and 9999999
+        {
+            "response_bytea_output": {test_string, "hex"}
+        },  # config option is one of `escape` or `hex`
+        {
+            "vacuum_autovacuum_analyze_scale_factor": {"-1", "0"}
+        },  # config option is between 0 and 100
+        {
+            "vacuum_autovacuum_vacuum_scale_factor": {"-1", "0"}
+        },  # config option is between 0 and 100
+        {
+            "vacuum_autovacuum_analyze_threshold": {"-1", "0"}
+        },  # config option is between 0 and 2147483647
+        {
+            "vacuum_autovacuum_freeze_max_age": {"99999", "100000"}
+        },  # config option is between 100000 and 2000000000
+        {
+            "vacuum_autovacuum_vacuum_cost_delay": {"-2", "-1"}
+        },  # config option is between -1 and 100
+        {
+            "vacuum_vacuum_freeze_table_age": {"-1", "0"}
+        },  # config option is between 0 and 2000000000
+    ]
+
+    for config in configs:
+        for k, v in config.items():
+            await ops_test.model.applications[DATABASE_APP_NAME].set_config({k: list(v)[0]})
+            await ops_test.model.wait_for_idle(
+                apps=[DATABASE_APP_NAME], status="blocked", timeout=1500
+            )
+            assert "Configuration Error" in leader_unit.workload_status_message
+
+    config = {}
+    for c in configs:
+        for k, v in c.items():
+            config.update({k: list(v)[1]})
+            break
+
+    await ops_test.model.applications[DATABASE_APP_NAME].set_config(config)
+    await ops_test.model.wait_for_idle(apps=[DATABASE_APP_NAME], status="active", timeout=1500)
```

Comment:
Once `v` is list:
```suggestion
    charm_config = {}
    for config in configs:
        for k, v in config.items():
            logger.info(k)
            charm_config[k] = v[0]
            await ops_test.model.applications[DATABASE_APP_NAME].set_config(charm_config)
            await ops_test.model.wait_for_idle(
                apps=[DATABASE_APP_NAME], status="blocked", timeout=100
            )
            assert "Configuration Error" in leader_unit.workload_status_message
            charm_config[k] = v[1]

    await ops_test.model.applications[DATABASE_APP_NAME].set_config(charm_config)
    await ops_test.model.wait_for_idle(apps=[DATABASE_APP_NAME], status="active", timeout=100)
```
We should test each invalid value one by one, since otherwise the other values would keep the charm blocked, even if the validation didn't work for one specific value.

We should log the value that is currently tested, so it would be easier to debug in the future.

We don't have to wait for idle long, on_config should be fairly fast.